bayesian/utility/v2/ca_response_builder.py

```diff
@@ -159,26 +159,21 @@ def get_version_without_cves(self, latest_non_cve_versions):
 
         :return: Highest Version out of all latest_non_cve_versions.
         """
-        logger.debug("All Versions with Vulnerabilities.")
-        input_version_tuple = version_info_tuple(
-            convert_version_to_proper_semantic(self.version)
-        )
         highest_version = ''
-        for version in latest_non_cve_versions:
-
-            graph_version_tuple = version_info_tuple(
-                convert_version_to_proper_semantic(version)
-            )
-            if graph_version_tuple > input_version_tuple:
-                if not highest_version:
-                    highest_version = version
-                highest_version_tuple = version_info_tuple(
-                    convert_version_to_proper_semantic(highest_version)
-                )
-                # If version to recommend is closer to what a user is using then, use less than
-                # If recommendation is to show highest version then, use greater than
-                if graph_version_tuple > highest_version_tuple:
-                    highest_version = version
+        try:
+            input_version_comparable = ComparableVersion(self.version)
+
+            # latest non-cve is list with only one entry.
+            latest_non_cve_version_comparable = ComparableVersion(latest_non_cve_versions[0])
+
+            if latest_non_cve_version_comparable > input_version_comparable:
+                highest_version = latest_non_cve_versions[0]
+
+        except Exception as e:
+            logger.error(f"Package {self.package} @ {self.version} raised exception {e}")
```

Comment:
It is not recommended to use preformatted string in logging. it would expanded regardless of log level which is not good. 
```suggestion
            logger.exception("Unable to parse version %s@%s", self.package, self.version)
```